pass-data-client/src/main/java/org/dataconservancy/pass/client/elasticsearch/ElasticsearchPassClient.java

```diff
@@ -0,0 +1,225 @@
+        if (modelClass==PassEntity.class) {throw new IllegalArgumentException("modelClass cannot be the abstract class 'PassEntity.class'");}
+        if (attribute==null || attribute.length()==0) {throw new IllegalArgumentException("attribute cannot be null or empty");}
+        if (value==null) {throw new IllegalArgumentException("value cannot be null or empty");}
+        if (offset < 0) {throw new IllegalArgumentException("The offset value cannot be less than 0");}
+        if (limit < 0) {throw new IllegalArgumentException("The limit value cannot be less than 0");}
+                
+        String indexType = null;
+        
+        if (PassEntityType.getTypeByName(modelClass.getSimpleName())!=null) {
+            indexType = PassEntityType.getTypeByName(modelClass.getSimpleName()).getName();
+        }
+        
+        String attribs = String.format(QS_ATTRIB_TEMPLATE, attribute, value.toString());
+        String querystring = String.format(QS_TEMPLATE, indexType, attribs);                
+        Set<URI> passEntityUris = getIndexerResults(querystring, limit, offset);
+        
+        return passEntityUris;
+    }
+
+    
+    /**
+     * @see org.dataconservancy.pass.client.PassClient#findAllByAttributes(Class, Map<String, Object>)
+     */
+    public <T extends PassEntity> Set<URI> findAllByAttributes(Class<T> modelClass, Map<String, Object> valueAttributesMap) {
+        return findAllByAttributes(modelClass, valueAttributesMap, ElasticsearchConfig.getIndexerLimit(), 0);
+    }
+    
+    
+    /**
+     * @see org.dataconservancy.pass.client.PassClient#findAllByAttributes(Class, Map<String, Object>, int, int)
+     */
+    public <T extends PassEntity> Set<URI> findAllByAttributes(Class<T> modelClass, Map<String, Object> valueAttributesMap, int limit, int offset) {
+        if (modelClass==null) {throw new IllegalArgumentException("modelClass cannot be null");}
+        if (modelClass==PassEntity.class) {throw new IllegalArgumentException("modelClass cannot be the abstract class 'PassEntity.class'");}
+        if (valueAttributesMap==null || valueAttributesMap.size()==0) {throw new IllegalArgumentException("valueAttributesMap cannot be empty");}
+        if (offset < 0) {throw new IllegalArgumentException("The offset value cannot be less than 0");}
+        if (limit < 0) {throw new IllegalArgumentException("The limit value cannot be less than 0");}
+
+        LOG.debug("Searching for {} using multiple filters", modelClass.getSimpleName());
+        
+        String indexType = null;
+        
+        if (PassEntityType.getTypeByName(modelClass.getSimpleName())!=null) {
+            indexType = PassEntityType.getTypeByName(modelClass.getSimpleName()).getName();
+        }
+        
+        StringBuilder attribs = new StringBuilder("");
+        for(Entry<String,Object> attr : valueAttributesMap.entrySet()) {
+            attribs.append(String.format(QS_ATTRIB_TEMPLATE, attr.getKey(), attr.getValue().toString()));
```

Comment:
Ah, I see.  So the `Object` in the map ultimately gets turned into a string.  There should probably be a check so that an exception is thrown if the object is a `Collection` subclass.  Clearly, single-values are needed here.